test/e2e/framework.go

```diff
@@ -895,11 +895,14 @@ func getImageName(uri string) string {
 // createPodOnNode creates a pod in the test namespace with a container whose type is decided by imageName.
 // Pod will be scheduled on the specified Node (if nodeName is not empty).
 // mutateFunc can be used to customize the Pod if the other parameters don't meet the requirements.
-func (data *TestData) createPodOnNode(name string, nodeName string, image string, command []string, args []string, env []corev1.EnvVar, ports []corev1.ContainerPort, hostNetwork bool, mutateFunc func(*corev1.Pod)) error {
+func (data *TestData) createPodOnNode(name string, ns string, nodeName string, image string, command []string, args []string, env []corev1.EnvVar, ports []corev1.ContainerPort, hostNetwork bool, mutateFunc func(*corev1.Pod)) error {
 	// image could be a fully qualified URI which can't be used as container name and label value,
 	// extract the image name from it.
 	imageName := getImageName(image)
-	return data.createPodOnNodeInNamespace(name, testNamespace, nodeName, imageName, image, command, args, env, ports, hostNetwork, mutateFunc)
+	if ns == "" {
+		ns = testNamespace
+	}
```

Comment:
is there any value in letting the user pass an empty string? You have already updated all the call sites to use `testNamespace` explicitly as far as I can tell

I think we could get rid of all the occurrences of this check

Reply:
I added this check in case user pass an empty string in the future use. Do you think there's no such a need? I can remove those if so.

Reply:
I think we should just let it fail if it's the empty string / an invalid namespace. We don't need to add that code everywhere IMO.

Reply:
I think Antonin's suggestion applies to every `namespace == ""` check? Then this one should be updated as well:
https://github.com/antrea-io/antrea/pull/2425/files#diff-85ebf4a94b7ed92e12d4ccea5e41be34ceeaf112df1ebba49de2fec7c297b780R1079

Reply:
> I think Antonin's suggestion applies to every `namespace == ""` check? Then this one should be updated as well:
> https://github.com/antrea-io/antrea/pull/2425/files#diff-85ebf4a94b7ed92e12d4ccea5e41be34ceeaf112df1ebba49de2fec7c297b780R1079

Good catch. Thanks for reminding me.